source/agora/test/Base.d

```diff
@@ -1028,3 +1034,77 @@ public bool containSameBlocks (APIS)(APIS nodes, size_t height)
 
     return true;
 }
+
+
+/*******************************************************************************
+
+    Generate a genesis block.
+
+    For each key pair, a freeze transaction and an Enrollment
+    will be created and added to the generated Genesis block.
+    The first transaction is the `UnitTestGenesisTransaction`
+    The minimum number of enrollments is Enrollment.MinValidatorCount.
+
+    Params:
+        key_pairs = key pairs for signing enrollments with
+
+    Returns:
+        The genesis block
+
+*******************************************************************************/
+
+private immutable(Block) makeGenesisBlock (in KeyPair[] key_pairs)
+{
+    import agora.common.Serializer;
+
+    // 1 payment tx, the rest are freeze txs
+    Transaction[] txs;
+    txs ~= UnitTestGenesisTransaction.serializeFull.deserializeFull!Transaction;
+    Enrollment[] enrolls;
+
+    foreach (idx; 0 .. max(Enrollment.MinValidatorCount, key_pairs.length))
+    {
+        auto key_pair = (idx < key_pairs.length)
+                        ? key_pairs[idx]
+                        : KeyPair.random;
+
+        Transaction tx =
+        {
+            type : TxType.Freeze,
+            inputs : [ Input.init ],
+            outputs : [Output(Amount.MinFreezeAmount, key_pair.address)]
+        };
+
+        txs ~= tx;
+        Hash txhash = hashFull(tx);
+        Hash utxo = UTXOSet.getHash(txhash, 0);
+        scope enr = new EnrollmentManager(":memory:", key_pair);
+        scope (exit) enr.shutdown();
+
+        Enrollment enroll;
+        const StartHeight = 1;
+        assert(enr.createEnrollment(utxo, StartHeight, enroll));
+        enrolls ~= enroll;
+    }
+
+    txs.sort;
+    Hash[] merkle_tree;
+    auto merkle_root = Block.buildMerkleTree(txs, merkle_tree);
+
+    immutable(BlockHeader) makeHeader ()
+    {
+        return immutable(BlockHeader)(
+            Hash.init,   // prev
+            0,           // height
+            merkle_root,
+            BitField!uint.init,
+            Signature.init,
+            enrolls.assumeUnique,
```

Comment:
enrolls.sort!("a.utxo_key > b.utxo_key");
The enrollments should be aligned like this.

Reply:
Thank you for your good suggestion.